src/rest_vol_dataset.c

```diff
@@ -4137,4 +4165,201 @@ rv_dataset_write_cb(hid_t mem_type_id, hid_t mem_space_id, void *buf, struct res
+        case H5S_SEL_ERROR:
+        case H5S_SEL_N:
+        case H5S_SEL_NONE:
+        default:
+            FUNC_GOTO_DONE(TRUE);
+    } /* end switch */
+
+done:
+    if (block)
+        RV_free(block);
+    if (count)
+        RV_free(count);
+    if (dims)
+        RV_free(dims);
+    if (stride)
+        RV_free(stride);
+    if (start)
+        RV_free(start);
+
+    return ret_value;
+} /* end RV_dataspace_selection_is_contiguous() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_convert_start_to_offset
+ *
+ * Purpose:     Convert starting position value to an offset value.
+ *
+ * Return:      Offset value on success/Negative value on failure.
+ *
+ * Programmer:  Jan-Willem Blokland
+ *              August, 2023
+ */
+static hssize_t
+RV_convert_start_to_offset(hid_t space_id)
+{
+    hsize_t *dims      = NULL;
+    hsize_t *start     = NULL;
+    hsize_t *end       = NULL;
+    hssize_t ret_value = 0;
+    int      ndims, i;
+
+    switch (H5Sget_select_type(space_id)) {
+        case H5S_SEL_HYPERSLABS: {
+            if ((ndims = H5Sget_simple_extent_ndims(space_id)) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, -1, "can't retrieve dataspace dimensionality");
+
+            if (NULL == (dims = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*dims))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, -1,
+                                "can't allocate space for dimension 'dims' values");
+
+            if (H5Sget_simple_extent_dims(space_id, dims, NULL) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, -1, "can't get dataspace dimension size");
+
+            if (NULL == (start = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*start))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, -1,
+                                "can't allocate space for hyperslab selection 'start' values");
+            if (NULL == (end = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*end))))
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, -1,
+                                "can't allocate space for hyperslab selection 'end' values");
+
+            if (H5Sget_select_bounds(space_id, start, end) < 0)
+                FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, -1,
+                                "can't get bounding box of hyperslab selection");
+
+            ret_value = start[0];
+            for (i = 1; i < ndims; i++) {
+                ret_value = ret_value * dims[i] + start[i];
+            }
+            break;
+        } /* H5S_SEL_HYPERSLABS */
+
+        case H5S_SEL_POINTS:
+            FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_BADVALUE, -1,
+                            "for point selection, computing the offset is not supported");
+            break;
+
+        case H5S_SEL_ALL:
+        case H5S_SEL_ERROR:
+        case H5S_SEL_N:
+        case H5S_SEL_NONE:
+        default:
+            ret_value = 0;
```

Comment:
Same comment here about H5S_SEL_ERROR and H5S_SEL_N making this function return a failure value